Prolangle/Extensions/LanguageEnumerablePickExtensions.cs

```diff
@@ -0,0 +1,32 @@
+using Prolangle.Languages.Framework;
+using Prolangle.Services;
+using Weighted_Randomizer;
+
+namespace Prolangle.Extensions;
+
+public static class LanguageEnumerablePickExtensions
+{
+	public static ILanguage PickWeightedRandom(this IEnumerable<ILanguage> languages,
+		GameSeeder seeder)
+	{
+		var randomizer = new StaticWeightedRandomizer<ILanguage>(seeder.Seeder());
+
+		foreach (ILanguage language in languages)
+			randomizer.Add(language, GetWeightFromTiobeRating(language.TiobeRating));
+
+		return randomizer.NextWithReplacement();
+	}
+
+	/// <summary>
+	/// We weigh a language's TIOBE rating against the frequency in
+	/// which the language should appear. I.e., _more commonly_-used
+	/// languages become the language to guess _less_ often.
+	/// </summary>
+	internal static int GetWeightFromTiobeRating(double? tiobeRating)
+	{
+		if (tiobeRating is null or 0)
+			return 100;
+
+		return (int)Math.Min(100.0 / tiobeRating.Value, 99.0);
```

Comment:
Yeah... This just caps the weight, which is also a solution, but ideally we'd scale the highest weight down so that it's in the range 99-0.

Reply:
I'm not quite sure what you mean.

Reply:
It maps from 0 to 100, but we have no languages with a rating of 100. Ideally we map the values from 0 to (max rating of all available languages) to 0 to 99 so that we actually use the whole range between 0 and 99 and not just 0 to ~20.

Reply:
So, something like `var scaleFactor = 99.0 / languages.Max(l => l.TiobeRating)`?

Reply:
Jo